api/envoy/config/core/v3/address.proto

```diff
@@ -38,7 +38,6 @@ message EnvoyInternalAddress {
   oneof address_name_specifier {
     option (validate.required) = true;
 
-    // [#not-implemented-hide:] The :ref:`listener name <envoy_v3_api_field_config.listener.v3.Listener.name>` of the destination internal listener.
```

Comment:
Please keep the comment describing the field.

Reply:
This comment is not true, address has nothing to do with the listener name.

Reply:
IIUC the value here needs to point to another listener.
Would it be possible to add a reference to the field that this value will be matched to?

Reply:
seems like it changed what the value expect to from the beginning. Maybe worth to ask again whether this field still name correctly, this can be a chance to change the name before expose to the end user.

Reply:
Yes, I agree we need to fix this. I think we should use a dedicated field in InternalListener instead of internal address in the generic `address` field in Listener. WDYT?

Reply:
The simplest thing is to just really use the listener name. That keeps LDS to a minimum and I can't think of any reason why more flexibility would be needed here (multiple addresses don't work anyways, but can always be added to `InternalListener`).

Reply:
I'm not sure the usecase of using multiple internal addresses, it seems useless. I guess the multiple addresses usecase is internal address mix with IP address. From this point of view, just use the listener name should be fine.

Reply:
> I'm not sure the usecase of using multiple internal addresses, it seems useless. I guess the multiple addresses usecase is internal address mix with IP address. From this point of view, just use the listener name should be fine.

suddenly found this may be the multiple internal addresses usecase https://github.com/envoyproxy/envoy/pull/22274/files#r935253240